addons/resource/models/utils.py

```diff
@@ -0,0 +1,41 @@
+from odoo.osv.expression import normalize_domain, NOT_OPERATOR, DOMAIN_OPERATORS
+
+def filter_domain_leaf(domain, leaf_check):
+    """
+    filter_domain_lead only keep the leaves of a domain that verify a given check. Logical operators that involves
+    a leaf that is undetermined (because it does not pass the check) are ignored.
+    
```

Comment:
```suggestion

```
There are some trailing spaces 🙂 
That is why the ci/style is not happy

Perhaps your editor can automatically remove the trailing spaces once you save? I can do that with mine. 🙂